File: java/339A.java
// 339 A. Helpful Maths

import java.util.*;

public class Main {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        String s = sc.next();
        String[] parts = s.split("\\+");

        int[] nums = new int[parts.length];
        for (int i = 0; i < parts.length; i++) {
            nums[i] = Integer.parseInt(parts[i]);
        }

        Arrays.sort(nums);

        for (int i = 0; i < nums.length; i++) {
            System.out.print(nums[i]);
            if (i != nums.length - 1)
                System.out.print("+");
        }

        sc.close();
    }
}


//testcases
// 2+4+1+5+2+6+9

File: cpp/339A.cpp
// 339 A. Helpful Maths

#include <bits/stdc++.h>
using namespace std;

int main() {
    string s;
    cin >> s;
    vector<int> nums;

    for (int i = 0; i < s.length(); i += 2) {
        nums.push_back(s[i] - '0');
    }

    sort(nums.begin(), nums.end());

    for (int i = 0; i < nums.size(); i++) {
        cout << nums[i];
        if (i != nums.size() - 1)
            cout << "+";
    }

    return 0;
}


// testcases
// 2+4+1+5+2+6+9